### ISSUE TITLE
Using iteration counter

### ISSUE THREAD

**Issue by Steve-Tod** [xinntao/BasicSR#77]
Hi,
I was wondering if it is reasonable to use iteration as the counter. If I increase the batch size, the model will have processed more images at the same iteration. Thus, the learning rate milestone actually changes with the batch size. Maybe you should use iteration*batch_size as the counter?

**Comment by xinntao** [comment]
If you increase batch sizes, you should also 1) increase the learning rate; 2) decrease the learning rate milestones (as what you said), to make these two situations be (almost) the same. 

Usually, people use epoch or iter as the counter. But the epoch will be very different for different datasets. So I choose to use iter as the counter. 

Things will be more complicated when you use multi-GPUs (change the batch size), or change the training patch size. So at this time, I choose one setting and it may not be roubust to other settings. Maybe you could change it if you change the batch size.

**Comment by Steve-Tod** [comment]
Thanks for your reply!
I also got another question.
`l_g_gan = self.l_gan_w * (self.cri_gan(pred_d_real - torch.mean(pred_g_fake), False) +
          self.cri_gan(pred_g_fake - torch.mean(pred_d_real), True)) / 2`
Why do you minus the mean of pred_g_fake and pred_d_real?

**Comment by xinntao** [comment]
This is the implementation of RaGAN, you can refer to the original paper. https://arxiv.org/abs/1807.00734

**Comment by Steve-Tod** [comment]
Oh, thank you! Last question😂
When you were trained your own ESRGAN for the workshop, did you also extract sub-images from Flickr dataset? Or you just use the low-resolution images of the dataset directly. 
Because after I downloaded the Flickr dataset you provided, I noticed there is already a Flickr2K_LR_bicubic folder and it contains the same amount of pictures as the Flickr2K_HR folder.

**Comment by xinntao** [comment]
I extracted sub-images from Flickr dataset.

I think Flickr2K_LR_bicubic and Flickr2K_HR are paired images, used for training, which is different from sub-images.

**Comment by Steve-Tod** [comment]
Thanks!

**Comment by TomArrow** [comment]
Does changing batch size from 16 to 20 already require a change in the learning rate or will I be okay?

**Comment by xinntao** [comment]
@TomArrow 
Empirically, if you use a larger batch size, you can use a larger learning rate. You can try a linear amplification ratio.

**Comment by Zency-Sun** [comment]
Following on from the above question, I would like to discuss with you how the change in Batch relates to the change in learning rate and the number of iterations? For example, here is an example:
- With Batch=32, the iterations are 500K, lr=2e-4 and the milestones are [250K, 400K, 450K, 475K].
- With Batch=16, the iterations are 1000K, lr=1e-4 and the milestones are [500K, 800K, 900K, 950K].

This case is linear. When Batch changes, do the learning rate and the number of iterations have to follow a linear relationship with the change in Batch, or is it a square root or exponential relationship?
